docs/networking.md

```diff
@@ -0,0 +1,134 @@
+```
+onboot:
+  - name: dhcpcd
+    image: linuxkit/dhcpcd:<hash>
+    command: ["/sbin/dhcpcd", "--nobackground", "-f", "/dhcpcd.conf", "-1"]
+```
+
+The alternative method is to make it a service. This ensures dhcpcd catches any updates,
+but may cause issues with services that expect an IP address on startup,
+since service launch order is *not* guaranteed.
+
+Similarly, using WiFi requires wpa_supplicant in addition to dhcpd.
+
+Finally, if you require any specialized firmware or device drivers for your network card,
+you may need to load it.
+
+While the common setup of LinuxKit doesn't usually bring any network connectivity issues,
+they still may occur. Moreover, LinuxKit may be used on the real hardware,
+and in that case the network setup could be a major part of the getting the image run.
+
+## General Steps
+
+There is a standard checklist one have to follow to set up the networking:
+
+1. Drivers are loaded by kernel (otherwise, may need a [custom kernel](kernels.md));
+2. Firmware is loaded (otherwise, need to put firmware blobs into the image);
+3. Network interfaces are visible in the system (otherwise, need to revisit items 1 and 2);
+4. Wi-Fi: wpa_supplicant is running (otherwise, need to include it into services section);
+5. DHCP: dhcpcd service is loaded.
+6. May need to set up static IP addressed and routes, but it's out of scope of this document.
+
+Usually, `lspci` provides the information about the detected devices, and
+`dmesg` prints the information whether the drivers for those devices
+are loaded or not (usually if nothing mentioned in the log, then no driver is loaded).
+
+### Wi-Fi Notes
+
+Vanilla LinuxKit kernel doesn't provide the Wi-Fi support, therefore
+the custom kernel should be used. The major options to enable:
+ - `CONFIG_WIRELESS=y` (Wireless);
+ - `CONFIG_CFG80211=y` (Improved wireless configuration API);
+ - `CONFIG_MAC80211=y` (Generic IEEE 802.11 Networking Stack);
+ - `CONFIG_WLAN=y` (Wireless LAN).
+
+`wpa_supplicant` requires CONFIG_CFG80211 to configure the Wi-Fi connections.
+CONFIG_CFG80211_WEXT may be useful too for the old hardware.
+
+The particular driver is better to compile as a module (and it's mandatory if it
+requires a userspace firmware), and enable it with the modprobe later.
+Please note that enabling the driver family is not always enough, for example,
+Broadcom FullMAC device (`CONFIG_BRCMFMAC=m`) may need to have PCIE bus interface
+support (`CONFIG_BRCMFMAC_PCIE=y`) enabled too to work.
+
+In the image configuration please add the modprobe item
+(with the example of the Broadcom FullMAC device):
+```
+onboot:
+  - name: modprobe
+    image: linuxkit/modprobe:<hash>
+    command: ["modprobe", "-a", "brcmfmac"]
+```
+
+The next step is getting an optional firmware. Please use the official
+[Linux Wireless wiki](https://wireless.wiki.kernel.org/welcome) and search for the
+driver instructions there (`dmesg` may give the helpful messages too).
+
+### Userspace firmware
```

Comment:
I would move this maybe above the WiFi notes as this is generic, ie applies to other NICs as well (but mostly WiFi)